pkgs/development/libraries/gtk-sharp/3.0.nix

```diff
@@ -13,30 +13,29 @@
 , libxml2
 , monoDLLFixer
 }:
-
-stdenv.mkDerivation {
-  name = "gtk-sharp-2.99.3";
+let
+  pname = "gtk-sharp";
+  version = "2.99.3";
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
 
   builder = ./builder.sh;
   src = fetchurl {
-    #"mirror://gnome/sources/gtk-sharp/2.99/gtk-sharp-2.99.3.tar.xz";
-    url = "http://ftp.gnome.org/pub/GNOME/sources/gtk-sharp/2.99/gtk-sharp-2.99.3.tar.xz";
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
     sha256 = "18n3l9zcldyvn4lwi8izd62307mkhz873039nl6awrv285qzah34";
   };
 
-  # patch bad usage of glib, which wasn't tolerated anymore
-  # prePatch = ''
-  #   for f in glib/glue/{thread,list,slist}.c; do
-  #     sed -i 's,#include <glib/.*\.h>,#include <glib.h>,g' "$f"
-  #   done
-  # '';
-
   nativeBuildInputs = [ pkgconfig ];
   buildInputs = [
     mono glib pango gtk3 GConf libglade libgnomecanvas
     libgtkhtml libgnomeui libgnomeprint libgnomeprintui gtkhtml libxml2
   ];
 
+  patchPhase = ''
+    substituteInPlace gtk/gui-thread-check/profiler/gui-thread-check.c \
```

Comment:
Maybe better use https://github.com/mono/gtk-sharp/pull/266

Reply:
@jtojnar Added that patch to the repo.

Reply:
Is it not possible to fetch it with `fetchpatch`?

Reply:
I did not know about fetchpatch, and I don't know how to get a patch file from this: https://github.com/mono/gtk-sharp/pull/266/files

Reply:
You can just add `.patch` after the URL, for example here you would fetch `https://github.com/mono/gtk-sharp/commit/c1b9bd4214b78dc426bc6e6a07c23a8ab4ab50ea.patch`

Reply:
wow, nice trick... I'll do a fetchpatch then.

Reply:
I noticed that when I `nix-prefetch-url https://github.com/mono/gtk-sharp/commit/401df51bc461de93c1a78b6a7a0d5adc63cf186c.patch` I get a different hash than the nix expression evaluator ... do you know why is that happening?

Reply:
`fetchpatch` also cleans the patch up so you would need a hash of the cleaned file.